upgraderesponder/service.go

```diff
@@ -49,12 +46,25 @@ var (
 )
 
 type Server struct {
-	done           chan struct{}
-	VersionMap     map[string]*Version
-	TagVersionsMap map[string][]*Version
-	influxClient   influxcli.Client
-	db             *maxminddb.Reader
-	dbCache        *DBCache
+	done chan struct{}
+	// The set of versions that is returned when the client does
+	// not include the information required to make an InstanceInfo.
+	DefaultVersions []rd.Version
+	// Maps Rules to a slice of versions with Version.Supported
+	// precomputed according to Rule.Constraints.
+	PrecomputedVersions []PrecomputedVersion
```

Comment:
The main reason I did this was to shift as many possible errors to startup time, rather than when a request comes in. That way, it's more likely they'll be caught and fixed. The optimization is just gravy.

Reply:
Ok, that is indeed a good reason!